src/Auth/OAuth/Bitbucket.php

```diff
@@ -14,6 +14,11 @@ class Bitbucket extends OAuth
      */
     protected $user = [];
 
+    /**
+     * @var array
+     */
+    protected $requiredScope = [];
+
```

Comment:
Instead of having a new attribute '$requiredScope' we can overwrite '$userScope'.

This will save you the writing of this loop in every adapter:
```
foreach ($this->requiredScope as $item) {
           $this->addScope($item);
 }
```

For example:

    /**
     * @var array
     */
    protected $userScope = ['email', 'job_title'];

Reply:
You can also name the abstract attribute 'userScope' with just **'scope'**

Reply:
can you elaborate how this works?

Reply:
Also, I have renamed the variable in OAuth from userScopes to scopes